File: backend/timebank/src/main/java/kookmin/software/capstone2023/timebank/domain/model/AccountProfile.java
package kookmin.software.capstone2023.timebank.domain.model;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Embeddable
@Data
@AllArgsConstructor
@NoArgsConstructor
public class AccountProfile {

    /**
     * 닉네임
     */
    @Column(nullable = true, updatable = true, length = 20)
    private String nickname;

    /**
     * 프로필 이미지 URL
     */
    @Column(nullable = true, updatable = true, length = 100)
    private String imageUrl;
}

File: src/main/java/kookmin/software/capstone2023/timebank/domain/model/AccountProfile.kt
package kookmin.software.capstone2023.timebank.domain.model

import jakarta.persistence.Column
import jakarta.persistence.Embeddable

@Embeddable
class AccountProfile(
    /**
     * 닉네임
     */
    @Column(nullable = true, updatable = true, length = 20)
    var nickname: String,

    /**
     * 프로필 이미지 URL
     */
    @Column(nullable = true, updatable = true, length = 100)
    var imageUrl: String,
)
